core/src/main/scala/cats/data/IorT.scala

```diff
@@ -0,0 +1,509 @@
+    def apply[A, B](test: Boolean, right: => B, left: => A)(implicit F: Applicative[F]): IorT[F, A, B] =
+      IorT(F.pure(if (test) Ior.right(right) else Ior.left(left)))
+  }
+
+  /**
+   * If the condition is satisfied, return the given `B` in `Ior.Right`, otherwise, return the given
+   * `A` in `Ior.Left`, lifted into the specified `Applicative`.
+   * {{{
+   * scala> import cats.data.IorT
+   * scala> import cats.implicits._
+   * scala> val userInput = "hello world"
+   * scala> IorT.cond[Option](
+   *      |   userInput.forall(_.isDigit) && userInput.size == 10,
+   *      |   userInput,
+   *      |   "The input does not look like a phone number")
+   * res0: cats.data.IorT[Option,String,String] = IorT(Some(Left(The input does not look like a phone number)))
+   * }}}
+   */
+  final def cond[F[_]]: CondPartiallyApplied[F] = new CondPartiallyApplied[F]
+
+  /**
+   * If the condition is satisfied, return the value of `IorT.right` on `F[B]`, otherwise, return the
+   * value of `IorT.left` on `F[A]`.
+   * {{{
+   * scala> import cats.data.IorT
+   * scala> import cats.implicits._
+   * scala> val userInput = "hello world"
+   * scala> IorT.condF[Option, String, String](
+   *      |   userInput.forall(_.isDigit) && userInput.size == 10,
+   *      |   Some(userInput),
+   *      |   None)
+   * res0: cats.data.IorT[Option,String,String] = IorT(None)
+   * }}}
+   */
+  final def condF[F[_], A, B](test: Boolean, right: => F[B], left: => F[A])(implicit F: Functor[F]): IorT[F, A, B] =
+    IorT(if (test) F.map(right)(Ior.right) else F.map(left)(Ior.left))
+}
+
+private[data] abstract class IorTInstances extends IorTInstances1 {
+
+  implicit def catsDataShowForIorT[F[_], A, B](implicit sh: Show[F[Ior[A, B]]]): Show[IorT[F, A, B]] =
+    Contravariant[Show].contramap(sh)(_.value)
+
+  implicit def catsDataBifunctorForIorT[F[_]](implicit F: Functor[F]): Bifunctor[IorT[F, ?, ?]] =
+    new Bifunctor[IorT[F, ?, ?]] {
+      override def bimap[A, B, C, D](iort: IorT[F, A, B])(fa: A => C, fb: B => D): IorT[F, C, D] = iort.bimap(fa, fb)
+    }
+
+  implicit def catsDataTraverseForIorT[F[_], A](implicit F: Traverse[F]): Traverse[IorT[F, A, ?]] =
+    new IorTTraverse[F, A] { val F0: Traverse[F] = F }
+
+  implicit def catsDataMonoidForIorT[F[_], A, B](implicit F: Monoid[F[Ior[A, B]]]): Monoid[IorT[F, A, B]] =
+    new IorTMonoid[F, A, B] { val F0: Monoid[F[Ior[A, B]]] = F }
+}
+
+private[data] abstract class IorTInstances1 extends IorTInstances2 {
+  implicit def catsDataSemigroupForIorT[F[_], A, B](implicit F: Semigroup[F[Ior[A, B]]]): Semigroup[IorT[F, A, B]] =
+    new IorTSemigroup[F, A, B] { val F0: Semigroup[F[Ior[A, B]]] = F }
+
+  implicit def catsDataFoldableForIorT[F[_], A](implicit F: Foldable[F]): Foldable[IorT[F, A, ?]] =
+    new IorTFoldable[F, A] { val F0: Foldable[F] = F }
+
+  implicit def catsDataMonadErrorFForIorT[F[_], A, E](implicit FE: MonadError[F, E], A: Semigroup[A]): MonadError[IorT[F, A, ?], E] =
```

Comment:
yeah, +1 on switching the order of these two MonadError instance to be consistent with EitherT